File: tests/test-data/bad/go/crypto.go
package main

import (
	"crypto/md5"
	"crypto/sha1"
	"math/rand"
)

// BAD: go-crypto-001 — MD5 hash
// ruleid: go-crypto-001
func hashWithMD5(data []byte) []byte {
	h := md5.New()
	h.Write(data)
	return h.Sum(nil)
}

// ruleid: go-crypto-001
func checksumMD5(data []byte) [16]byte {
	return md5.Sum(data)
}

// BAD: go-crypto-001 — SHA1 hash
// ruleid: go-crypto-001
func hashWithSHA1(data []byte) []byte {
	h := sha1.New()
	h.Write(data)
	return h.Sum(nil)
}

// BAD: go-crypto-002 — math/rand for token generation
// ruleid: go-crypto-002
func generateToken() int {
	return rand.Intn(1000000)
}

// ruleid: go-crypto-002
func generateSessionID() int64 {
	return rand.Int63()
}

File: tests/test-data/bad/js/crypto.js
const crypto = require('crypto');

// BAD: js-crypto-001 — MD5 hash
// ruleid: js-crypto-001
const hashMd5 = crypto.createHash('md5').update(data).digest('hex');

// BAD: js-crypto-001 — SHA1 hash
// ruleid: js-crypto-001
const hashSha1 = crypto.createHash('sha1').update(data).digest('hex');

// BAD: js-crypto-002 — Math.random for token
// ruleid: js-crypto-002
const token = Math.random().toString(36).substring(2);

// BAD: js-crypto-002 — Math.random for secret
// ruleid: js-crypto-002
const secret = Math.random().toString(36);
